cpp/src/arrow/compute/kernels/scalar_set_lookup.cc

```diff
@@ -272,10 +278,9 @@ struct IsInVisitor {
 
   Status Visit(const DataType&) {
     const auto& state = checked_cast<const SetLookupState<NullType>&>(*ctx->state());
+    // XXX should skip_nulls be taken into account?
```

Comment:
It's a bit strange corner case, but for consistency probably yes? 

Right now for null array the skip_nulls argument has no effect (using this PR):

```
In [27]: pc.is_in(pa.array([None]), value_set=pa.array([None]), skip_null=False)
Out[27]: 
<pyarrow.lib.BooleanArray object at 0x7fac1d800880>
[
  true
]

In [28]: pc.is_in(pa.array([None]), value_set=pa.array([None]), skip_null=True)
Out[28]: 
<pyarrow.lib.BooleanArray object at 0x7fac1d800ca0>
[
  true
]
```
while if you add a non-null type to the array creation, but use the same values, you get a different result:

```
In [30]: pc.is_in(pa.array([None], type="int64"), value_set=pa.array([None], type="int64"), skip_null=False)
Out[30]: 
<pyarrow.lib.BooleanArray object at 0x7fac1d82f880>
[
  true
]

In [31]: pc.is_in(pa.array([None], type="int64"), value_set=pa.array([None], type="int64"), skip_null=True)
Out[31]: 
<pyarrow.lib.BooleanArray object at 0x7fac1d7c2040>
[
  false
]
```

Reply:
Good point.